report.php

```diff
@@ -5,8 +5,91 @@
  */
 
 require __DIR__ . '/functions.php';
+require __DIR__ . '/vendor/autoload.php';
 
 // Check required environment variables
 check_required_env();
 
-// @todo process phpunit test run results; send to WordPress.org
+$WPT_SSH_CONNECT = getenv( 'WPT_SSH_CONNECT' );
+$WPT_TEST_DIR = getenv( 'WPT_TEST_DIR' );
+$WPT_PREPARE_DIR = getenv( 'WPT_PREPARE_DIR' );
+$WPT_SSH_OPTIONS = getenv( 'WPT_SSH_OPTIONS' );
+
+log_message('Getting SVN Revision');
+$rev = exec('git -C ' . escapeshellarg( $WPT_PREPARE_DIR ) . ' log -1 --pretty=%B | grep "git-svn-id:" | cut -d " " -f 2 | cut -d "@" -f 2');
+
+log_message('Copying junit.xml results');
+$junit_location = escapeshellarg( $WPT_TEST_DIR ) . '/tests/phpunit/build/logs/junit.xml';
+
+if ( false !== $WPT_SSH_CONNECT ) {
+	$junit_location = '-e "ssh ' . $WPT_SSH_OPTIONS . '" ' . escapeshellarg( $WPT_SSH_CONNECT . ':' . $junit_location );
+}
+
+$junit_exec = 'rsync -rv ' . $junit_location . ' ./';
+perform_operations( array(
+	$junit_exec,
+) );
+
+log_message( 'Processing and uploading junit.xml' );
+$success = upload( process( './junit.xml' ), $rev );
+
+if ( $success ) {
+	log_message( 'Results successfully uploaded' );
+} else {
+	log_message( 'Error uploading results' );
+}
+
+function process( $path )
```

Comment:
Sure! Wasn't sure if we should put single use functions there, but I'll move it.